common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java

```diff
@@ -316,25 +318,48 @@ public AcquireTokenResult acquireTokenSilent(
         // subsequent CacheRecords represent other profiles (projections) of this principal in
         // other tenants. Those tokens will be 'sparse', meaning that their AT/RT will not be loaded
         final ICacheRecord fullCacheRecord = cacheRecords.get(0);
-
-        if (accessTokenIsNull(fullCacheRecord)
+        if (fullCacheRecord.getAccessToken().refreshOnIsActive()){
+            Logger.info(
+                    TAG,
+                    "RefreshIn is active. This will extend your token usage in the rare case servers are not available."
```

Comment:
Should we throttle this log message?  The bigger ask would be to insure that some form of logging follows the token request around, so that in every place where we deal with fetching an access token we know that it is an eager refresh.

Reply:
Ok. I also added some logging to the CommandDispatcher.performRefresh(), and included correlationIds for the initial/silent and subsequent/refresh request.